api/hypershift/v1beta1/azure.go

```diff
@@ -98,6 +98,26 @@ type AzureNodePoolPlatform struct {
 	// If not specified, then Boot diagnostics will be disabled.
 	// +optional
 	Diagnostics *Diagnostics `json:"diagnostics,omitempty"`
+
+	// resourceTags is an optional list of additional tags to apply to Azure node
+	// instances.
+	//
+	// These will be merged with HostedCluster scoped tags, which take precedence in case of conflicts.
+	// These take precedence over tags defined out of band (i.e., tags added manually or by other tools outside of HyperShift) in Azure in case of conflicts.
+	//
+	// See https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-support and
+	// https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-resources for
+	// information on tagging Azure resources. Azure VMs support up to 50 tags
+	// per resource/resource group/subscription by default. Tag names can be up to 512 characters
+	// tag values can be up to 256 characters, and empty tag values are allowed.
+	// Kubernetes/OpenShift reserves 1 tag for its use,
+	// leaving up to 49 tags available for user use on instances.
+	//
+	// +kubebuilder:validation:MaxItems=49
+	// +listType=map
+	// +listMapKey=key
+	// +optional
+	ResourceTags []AzureResourceTag `json:"resourceTags,omitempty"`
```

Comment:
https://github.com/openshift/hypershift/pull/6846/files#r2364174798 still needs to be addressed